project_tag_security/views/project_project_views.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="edit_project" model="ir.ui.view">
+        <field name="name">project.project.form</field>
+        <field name="model">project.project</field>
+        <field name="inherit_id" ref="project.edit_project" />
+        <field name="arch" type="xml">
+            <field name="tag_ids" position="attributes">
+                <attribute
+                    name="domain"
+                >['|', ('allowed_project_ids', 'in', [id]),('allowed_project_ids', '=', False)]</attribute>
```

Comment:
Does this work?

```suggestion
                >[('allowed_project_ids', 'in', [False, id])]</attribute>
```

Reply:
I assumed you would suggest that but it doesn't work, with that domain you only get tags that have that project as allowed but not tags that have no project.